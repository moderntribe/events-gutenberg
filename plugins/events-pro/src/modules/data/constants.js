export const EVENTS_PRO_PLUGIN_NAME = 'events_pro';
