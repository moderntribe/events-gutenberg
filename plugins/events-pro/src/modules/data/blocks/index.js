/**
 * Internal dependencies
 */
import * as constants from './constants';
import reducer from './reducer';
import './sagas';

export default reducer;
export { constants };
