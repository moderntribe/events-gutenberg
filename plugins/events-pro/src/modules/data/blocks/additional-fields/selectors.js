/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);
export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);
