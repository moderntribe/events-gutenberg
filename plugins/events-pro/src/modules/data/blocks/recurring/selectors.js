/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getRules = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.recurring;
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getRules, getIndex ],
	( rules, index ) => rules[ index ],
);

export const getType = createSelector(
	[ getRule ],
	( rule ) => rule.type,
);

export const getAllDay = createSelector(
	[ getRule ],
	( rule ) => rule.all_day,
);

export const getMultiDay = createSelector(
	[ getRule ],
	( rule ) => rule.multi_day,
);

export const getStartDate = createSelector(
	[ getRule ],
	( rule ) => rule.start_date,
);

export const getStartTime = createSelector(
	[ getRule ],
	( rule ) => rule.start_time,
);

export const getEndDate = createSelector(
	[ getRule ],
	( rule ) => rule.end_date,
);

export const getEndTime = createSelector(
	[ getRule ],
	( rule ) => rule.end_time,
);

export const getBetween = createSelector(
	[ getRule ],
	( rule ) => rule.between,
);

export const getLimitType = createSelector(
	[ getRule ],
	( rule ) => rule.limit_type,
);

export const getLimit = createSelector(
	[ getRule ],
	( rule ) => rule.limit,
);

export const getDays = createSelector(
	[ getRule ],
	( rule ) => rule.days,
);

export const getWeek = createSelector(
	[ getRule ],
	( rule ) => rule.week,
);

export const getDay = createSelector(
	[ getRule ],
	( rule ) => rule.day,
);

export const getMonth = createSelector(
	[ getRule ],
	( rule ) => rule.month,
);

export const getTimezone = createSelector(
	[ getRule ],
	( rule ) => rule.timezone,
);
