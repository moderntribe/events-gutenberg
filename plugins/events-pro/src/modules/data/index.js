/**
 * External dependencies
 */
import reducer from './reducers';

import { actions } from '@moderntribe/common/data/plugins';
import { store } from '@moderntribe/common/store';
import { EVENTS_PRO_PLUGIN_NAME } from './constants';

export const initStore = () => {
	const { dispatch, injectReducers } = store;

	dispatch( actions.addPlugin( EVENTS_PRO_PLUGIN_NAME ) );
	injectReducers( { [ EVENTS_PRO_PLUGIN_NAME ]: reducer } );
};

export const getStore = () => store;
