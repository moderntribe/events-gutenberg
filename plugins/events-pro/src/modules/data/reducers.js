/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import blocks from './blocks';

export default combineReducers( { blocks } );
