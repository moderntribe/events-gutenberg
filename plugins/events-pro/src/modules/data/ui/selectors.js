/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
