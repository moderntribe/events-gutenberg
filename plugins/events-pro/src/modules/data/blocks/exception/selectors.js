/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	EXCEPTION_OCCURRENCE_OPTIONS,
} from './options';
import { SERIES_ENDS_OPTIONS } from '@moderntribe/events-pro/data/blocks/recurring/options';
import { constants } from '@moderntribe/common/data/plugins';

export const getExceptions = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.exception;
export const hasExceptions = createSelector( getExceptions, exceptions => !! exceptions.length );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getExceptions, getIndex ],
	( exceptions, index ) => exceptions[ index ],
);

export const getType = createSelector(
	[ getRule ],
	( exception ) => exception.type,
);

export const getAllDay = createSelector(
	[ getRule ],
	( exception ) => exception.all_day,
);

export const getMultiDay = createSelector(
	[ getRule ],
	( exception ) => exception.multi_day,
);

export const getStartDate = createSelector(
	[ getRule ],
	( exception ) => exception.start_date,
);

export const getStartTime = createSelector(
	[ getRule ],
	( exception ) => exception.start_time
);

export const getStartTimeNoSeconds = createSelector(
	[ getStartTime ],
	( startTime ) => startTime.slice( 0, -3 ),
);

export const getEndDate = createSelector(
	[ getRule ],
	( exception ) => exception.end_date,
);

export const getEndTime = createSelector(
	[ getRule ],
	( exception ) => exception.end_time
);

export const getEndTimeNoSeconds = createSelector(
	[ getEndTime ],
	( endTime ) => endTime.slice( 0, -3 ),
);

export const getBetween = createSelector(
	[ getRule ],
	( exception ) => exception.between,
);

export const getLimitType = createSelector(
	[ getRule ],
	( exception ) => exception.limit_type,
);

export const getLimit = createSelector(
	[ getRule ],
	( exception ) => exception.limit,
);

export const getDays = createSelector(
	[ getRule ],
	( exception ) => exception.days,
);

export const getWeek = createSelector(
	[ getRule ],
	( exception ) => exception.week,
);

export const getDay = createSelector(
	[ getRule ],
	( exception ) => exception.day,
);

export const getMonth = createSelector(
	[ getRule ],
	( exception ) => exception.month,
);

export const getTimezone = createSelector(
	[ getRule ],
	( exception ) => exception.timezone,
);

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( EXCEPTION_OCCURRENCE_OPTIONS, ( option ) => option.value === type )
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType )
);
