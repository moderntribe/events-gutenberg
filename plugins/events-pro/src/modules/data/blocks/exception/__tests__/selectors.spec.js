/**
 * External Dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

/**
 * Internal Dependencies
 */
import * as selectors from '../selectors';

describe( 'Exception Selectors', () => {
	let exception, state;
	beforeEach( () => {
		exception = {
			type: 'single',
		};
		state = {
			[ constants.EVENTS_PRO_PLUGIN ]: {
				blocks: {
					exception: [ exception ],
				},
			},
		};
	} );

	test( 'getExceptions', () => {
		expect( selectors.getExceptions( state ) ).toMatchSnapshot();
	} );
	test( 'getTypeOption', () => {
		expect( selectors.getTypeOption( state, { index: 0 } ) ).toMatchSnapshot();
	} );
} );
