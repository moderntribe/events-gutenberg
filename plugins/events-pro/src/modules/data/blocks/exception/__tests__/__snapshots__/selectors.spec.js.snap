// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exception Selectors getExceptions 1`] = `
Array [
  Object {
    "type": "single",
  },
]
`;

exports[`Exception Selectors getTypeOption 1`] = `
Object {
  "label": "Single Exception",
  "value": "single",
}
`;
